src/main/java/com/google/edith/QueryItems.java

```diff
@@ -0,0 +1,151 @@
+package com.google.edith;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.common.base.Splitter;
+import com.google.edith.servlets.Item;
+import com.google.edith.servlets.Receipt;
+import com.google.gson.Gson;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Finds the items on past receipts that have expired by the time of the user's next shopping trip.
+ */
+public class QueryItems {
+
+  private final DatastoreService datastore;
+  private UserService userService;
+
+  public QueryItems(DatastoreService datastore, UserService userService) {
+    this.datastore = datastore;
+    this.userService = userService;
+  }
+
+  public String findExpiredItems() {
+    List<Receipt> receipts = queryReceipts();
+    Set<Item> itemsToBuy = new HashSet<Item>();
+
+    for (Receipt receipt : receipts) {
+      double daysPassedSinceReceipt = findTimePassed(receipt.getDate());
+      Item[] items = receipt.getItems();
+
+      for (Item item : items) {
+        String expiration = item.expiration();
+        if (expiration.equals("NO_EXPIRATION")) {
+          continue;
+        }
+
+        List<String> expirationNumberAndUnit = Splitter.on(" ").splitToList(expiration);
+        double number = 0;
+        String unit = "";
+
+        for (String expirationPiece : expirationNumberAndUnit) {
+          try {
+            number = (double) Double.parseDouble(expirationPiece);
+          } catch (NumberFormatException e) {
+            unit = expirationPiece;
+          }
+        }
+        if (unit.toLowerCase().equals("weeks")) {
+          number = number * 7;
+        } else if (unit.toLowerCase().equals("months")) {
+          number = number * 30;
+        }
+
+        if (number < daysPassedSinceReceipt) {
+          itemsToBuy.add(item);
+        }
+      }
+    }
+
+    Gson gson = new Gson();
+    return gson.toJson(itemsToBuy);
+  }
+
+  /**
+   * Determines the number of days between when the receipt was stored in datastore and the current
+   * time.
+   */
+  private double findTimePassed(String receiptDateString) {
```

Comment:
Can we be a bit more descriptive with the name as I had to read the javadoc/code to find out. Time passed since when? Maybe findTimePassedSinceStored or findStoredDuration or findReceiptStorageDuration. Totally open since these are meh

Reply:
Also might even say findDurationDays or append Days to indicate the method returns days as units always